stubs/redis/redis/cluster.pyi

```diff
@@ -106,47 +106,49 @@ class RedisCluster(AbstractRedisCluster, RedisClusterCommands[_StrType], Generic
     def close(self) -> None: ...
 
 class ClusterNode:
-    host: Any
-    port: Any
-    name: Any
-    server_type: Any
-    redis_connection: Any
-    def __init__(self, host, port, server_type: Any | None = ..., redis_connection: Any | None = ...) -> None: ...
-    def __eq__(self, obj): ...
+    host: str
+    port: int
```

Comment:
Can `port` also be a `str`? In `__init__()`, it's passed to the `get_node_name()` function, and the `port` parameter of `get_node_name()` is annotated as `Union[str, int]`: https://github.com/redis/redis-py/blob/771109e74feb3b8f630c047bf4f93da2800ec22e/redis/cluster.py#L41

(The same question applies to the `port` parameter of `__init__` immediately below.)

Reply:
I think that it can not, because `node.port` is used in a couple of other places, for example: creating a `Redis(host, port)` instance, which is annotated to have `port` as `int`.

Source: https://github.com/redis/redis-py/blob/771109e74feb3b8f630c047bf4f93da2800ec22e/redis/cluster.py#L1370-L1378
Stub: https://github.com/python/typeshed/blob/c4ca110c5c02699350ca008050c430e23e087b93/stubs/redis/redis/client.pyi#L88

Reply:
Hmm... I think maybe our stub in `redis/client.pyi` should also be `port: int | str | None`. `redis.client.Redis` passes `port` to the constructor of `redis.connection.ConnectionPool`, which then passes port to the constructor of `redis.connection.Connection`. And `redis.connection.Connection` accepts `str` for the `port` parameter: https://github.com/redis/redis-py/blob/771109e74feb3b8f630c047bf4f93da2800ec22e/redis/connection.py#L539

Reply:
Let's leave this for the next PR 😉

Reply:
I will investigate where we can and cannot do it.

Reply:
Sounds good!